src/core/net/udp6.cpp

```diff
@@ -99,13 +99,26 @@ otError Udp::Socket::Bind(uint16_t aPort)
     return Bind(SockAddr(aPort));
 }
 
-#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
 otError Udp::Socket::BindToNetif(otNetifIdentifier aNetifIdentifier)
 {
-    return otPlatUdpBindToNetif(this, aNetifIdentifier);
-}
+    OT_UNUSED_VARIABLE(aNetifIdentifier);
+
+    otError error = OT_ERROR_NONE;
+
+#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
+    SuccessOrExit(error = otPlatUdpBindToNetif(this, aNetifIdentifier));
+#endif
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
```

Comment:
Change to `#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE` and add config check to see whether this is enabled when `OPENTHREAD_FTD` is not enabled in `config-check.h`?

Example: https://github.com/openthread/openthread/blob/master/src/core/config/openthread-core-config-check.h#L526

Reply:
This is indeed a good idea. But I prefer to do it in a separate PR because there are so many places that are using `#ifdef OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE` and they should all be fixed (but unrelated to this PR).

Reply:
SGTM.